ApplicationLibCode/ProjectDataModel/Surfaces/RimEnsembleSurface.cpp

```diff
@@ -121,8 +124,18 @@ void RimEnsembleSurface::loadDataAndUpdate()
         cvf::ref<RigSurface> firstSurface = sourceSurfaceForStatistics[0]->surfaceData();
 
         std::vector<cvf::ref<RigSurface>> sourceSurfaces;
-        for ( auto& w : sourceSurfaceForStatistics )
-            sourceSurfaces.push_back( RigSurfaceResampler::resampleSurface( firstSurface, w->surfaceData() ) );
+        {
+            int surfaceCount = static_cast<int>( sourceSurfaceForStatistics.size() );
+#pragma omp parallel for
+            for ( int i = 0; i < surfaceCount; i++ )
+            {
+                auto surf             = sourceSurfaceForStatistics[i];
+                auto resampledSurface = RigSurfaceResampler::resampleSurface( firstSurface, surf->surfaceData() );
+
+#pragma omp critical( sourceSurfaces )
```

Comment:
Pre-allocate the sourceSurface vector at correct size, and remove the critical section?